### PR TITLE
Update for CLAP 0.23

### PR DESCRIPTION
**Pull request by robbert-vdh**
CLAP 0.19 was very short lived because `clap_plugin_gui::create()` and `clap_plugin_gui::set_parent()` were one function which meant that setting the DPI scaling wasn't possible.

The changes were made based on this diff: https://github.com/free-audio/clap/compare/0.18.0...5d7484ebf93835755041a8f000bd32a1ddb914b5

### CODE REVIEW

**Comment by robbert-vdh** [comment]
I've updated this for CLAP 0.21 based on this diff: https://github.com/free-audio/clap/compare/5d7484ebf93835755041a8f000bd32a1ddb914b5..ac37c9ab03f26f639e1fc78cf29c6cab9cc65890

**Comment by robbert-vdh** [comment]
I've updated this for CLAP 0.23 based on this diff: https://github.com/free-audio/clap/compare/ac37c9ab03f26f639e1fc78cf29c6cab9cc65890..202c01d845eae72f2f9a1f6a4c4654b7b1145102

File: src/events.rs [hunk 1]
```diff
@@ -97,6 +96,16 @@ pub struct clap_event_param_mod {
...
 
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct clap_event_param_gesture {
+    pub header: clap_event_header,
+    pub param_id: clap_id,
+}
+
+unsafe impl Send for clap_event_param_gesture {}
+unsafe impl Sync for clap_event_param_gesture {}
```
Comment (micahrj) [inline src/events.rs]: I don't think these are necessary; everything in `clap_event_param_gesture` is already `Send + Sync`.

File: src/events.rs [hunk 2]
```diff
@@ -49,10 +49,9 @@ pub const CLAP_NOTE_EXPRESSION_VOLUME: clap_note_expression = 0;
 pub const CLAP_NOTE_EXPRESSION_PAN: clap_note_expression = 1;
 pub const CLAP_NOTE_EXPRESSION_TUNING: clap_note_expression = 2;
 pub const CLAP_NOTE_EXPRESSION_VIBRATO: clap_note_expression = 3;
+pub const CLAP_NOTE_EXPRESSION_EXPRESSION: clap_note_expression = 5;
 pub const CLAP_NOTE_EXPRESSION_BRIGHTNESS: clap_note_expression = 4;
```
Comment (micahrj) [inline src/events.rs]: I think these values are swapped (see https://github.com/free-audio/clap/blob/202c01d845eae72f2f9a1f6a4c4654b7b1145102/include/clap/events.h#L117-L118).

File: src/ext/gui.rs [hunk 3]
```diff
@@ -26,4 +67,5 @@ pub struct clap_host_gui {
     pub resize: unsafe extern "C" fn(host: *const clap_host, width: u32, height: u32) -> bool,
     pub request_show: unsafe extern "C" fn(host: *const clap_host),
     pub request_hide: unsafe extern "C" fn(host: *const clap_host),
+    pub closed: unsafe extern "C" fn(host: *const clap_host, was_destroyed: bool),
```
Comment (micahrj) [inline src/ext/gui.rs]: This struct is missing several changes which I believe I missed when updating to 0.18.0 (see https://github.com/free-audio/clap/blob/75dabe9c6c53192dc6249134c8bb9e562fa915a0/include/clap/ext/gui.h#L83-L98). Would you mind fixing them?

File: src/ext/gui.rs [hunk 4]
```diff
@@ -1,23 +1,64 @@
...
+pub type clap_xwnd = c_ulong;
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct clap_window {
+    pub api: *const c_char,
+    pub specific: clap_window_handle,
+}
+
+/// Defined as an anonymous union in [`clap_window` ]in the C-version.
```
Comment (micahrj) [inline src/ext/gui.rs]: Inconsequential nitpick: fix the space before `]` here?

**Review by micahrj — CHANGES_REQUESTED** [review]

File: src/events.rs (hunk 1 above)
Comment (robbert-vdh) [reply src/events.rs]: You're right, they are.

File: src/events.rs (hunk 2 above)
Comment (robbert-vdh) [reply src/events.rs]: Good catch. They changed order in CLAP 0.21 but I forgot to also swap the values around.

File: src/ext/gui.rs (hunk 3 above)
Comment (robbert-vdh) [reply src/ext/gui.rs]: Oops yeah I only went based on the diffs, I added the changes from https://github.com/free-audio/clap/commit/8fddd5b70352d01f9d5a1f23d4627a008d2feac4

File: src/ext/gui.rs (hunk 4 above)
Comment (robbert-vdh) [reply src/ext/gui.rs]: Fixed.

**Review by micahrj — APPROVED** [review]

**Pull request merged**